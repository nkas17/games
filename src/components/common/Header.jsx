import React from 'react';
import { Link, IndexLink } from 'react-router';

/**
 * Common Header component
 */
const Header = () => (
  <header>
    <h1>{"Welcome, let's Eat!"}</h1>
    <nav>
			<IndexLink to="/" activeClassName="active">Home</IndexLink>
			{" | "}
			<Link to="/about" activeClassName="active">About</Link>
		</nav>
  </header>
);

export default Header;