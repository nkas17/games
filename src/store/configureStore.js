import {createStore, applyMiddleware, compose} from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant' // eslint-disable-line
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const configureStore = (initialState) => {
	// Add wrapper to enable redux dev tools
	// @see https://github.com/zalmoxisus/redux-devtools-extension#12-advanced-store-setup

	const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(thunk, reduxImmutableStateInvariant()))
	);
}

export default configureStore;