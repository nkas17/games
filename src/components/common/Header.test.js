import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import configureStore from '../../store/configureStore';
import Header from './Header';

/* eslint-disable no-undef */

const store = configureStore();
const renderWithRouter = node => <Router>{node}</Router>;
const renderWithStore = node => <Provider store={store}>{node}</Provider>;

describe('Header', () => {
	it('renders correctly without loading', () => {
		const tree = renderer.create(
			renderWithStore(
				renderWithRouter(
					<Header
						loading={false}
					/>))).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly while loading', () => {
		const tree = renderer.create(
			renderWithStore(
				renderWithRouter(
					<Header
						loading
					/>))).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
