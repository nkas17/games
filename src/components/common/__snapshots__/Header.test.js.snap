// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly while loading 1`] = `
<header>
  <h1
    className="page-header"
  >
    come and eat!
  </h1>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      home
    </a>
     | 
    <a
      href="/recipe"
      onClick={[Function]}
    >
      recipes
    </a>
     | 
    <a
      href="/about"
      onClick={[Function]}
    >
      about
    </a>
  </nav>
</header>
`;

exports[`Header renders correctly without loading 1`] = `
<header>
  <h1
    className="page-header"
  >
    come and eat!
  </h1>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      home
    </a>
     | 
    <a
      href="/recipe"
      onClick={[Function]}
    >
      recipes
    </a>
     | 
    <a
      href="/about"
      onClick={[Function]}
    >
      about
    </a>
  </nav>
</header>
`;
